dags/main_summary.py

```diff
@@ -566,18 +567,42 @@
     output_visibility="private",
     dag=dag)
 
-taar_collaborative_recommender = EMRSparkOperator(
+taar_collaborative_recommender = SubDagOperator(
     task_id="addon_recommender",
-    job_name="Train the Collaborative Addon Recommender",
-    execution_timeout=timedelta(hours=10),
-    instance_count=20,
-    owner="mlopatka@mozilla.com",
-    email=["telemetry-alerts@mozilla.com", "mlopatka@mozilla.com", "vng@mozilla.com"],
-    env={"date": "{{ ds_nodash }}",
-         "privateBucket": "{{ task.__class__.private_output_bucket }}",
-         "publicBucket": "{{ task.__class__.public_output_bucket }}"},
-    uri="https://raw.githubusercontent.com/mozilla/telemetry-airflow/master/jobs/addon_recommender.sh",
-    dag=dag)
+    subdag=moz_dataproc_jar_runner(
+        parent_dag_name=dag.dag_id,
+        dag_name="addon_recommender",
+        job_name="Train the Collaborative Addon Recommender",
+        main_class="com.mozilla.telemetry.ml.AddonRecommender",
+        jar_urls=[
+            "https://s3-us-west-2.amazonaws.com/net-mozaws-data-us-west-2-ops-ci-artifacts"
+            "/mozilla/telemetry-batch-view/master/telemetry-batch-view.jar",
+        ],
+        jar_args=[
+          "--runDate={{ds_nodash}}",
+          "--inputTable=gs://moz-fx-data-derived-datasets-parquet/clients_daily/v6",
+          "--privateBucket=telemetry-parquet",
+          "--publicBucket=telemetry-public-analysis-2",
+        ],
+        cluster_name="addon-recommender-{{ds_nodash}}",
+        image_version="1.3",
+        worker_machine_type="n1-standard-8",
+        num_workers=20,
+        optional_components=[],
```

Comment:
nit:
```suggestion
```

Reply:
without this dataproc will be configured to install anaconda on startup, which we don't need.